smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -188,6 +188,40 @@ contract MixinTransfer is
     return fee.mul(transferFeeBasisPoints) / BASIS_POINTS_DEN;
   }
 
+  /**
+  * @notice Modify the expirationTimestamp of a key
+  * by a given amount.
+  * @param _owner The owner of the key to modify
+  * @param _deltaT The amount of time in seconds by which
+  * to modify the keys expirationTimestamp
+  * @param _addTime Choose whether to increase or decrease
+  * expirationTimestamp (false == decrease, true == increase)
+  * @dev Throws if owner does not have a valid key.
+  */
+  function _timeMachine(
+    address _owner,
+    uint256 _deltaT,
+    bool _addTime
+  ) internal
+    hasValidKey(_owner)
+  {
+    Key storage key = keyByOwner[_owner];
+    uint formerTimestamp = key.expirationTimestamp;
+    if(_addTime) {
+      if(formerTimestamp.add(_deltaT) > block.timestamp.add(expirationDuration)) {
+        key.expirationTimestamp = block.timestamp.add(expirationDuration);
```

Comment:
Hmm do we really want to cap transfers?  It seems if I own the time I should be able to transfer it all to a different account.

Reply:
Yes, you are right. I will modify this to allow adding an arbitrary amount of time.